### PR TITLE
Feature/orm frontend similarity vector search

### PR DESCRIPTION
**Pull request by keithCuniah**
ref #2029 
Create Vuex orm for vector similarity search

1/ add vector to the current dataset
2/ insert to a dedicated table the vector by recordId

Note: the Vector model is tottally dissociate of the current models orm But later in a refactoring the table will be link to the record table (we need to merge the weak labelling before that refactoring)
image

### CODE REVIEW

File: frontend/database/index.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ import { ObservationDataset } from "@/models/Dataset";
 import { Text2TextDataset } from "@/models/Text2Text";
 import { TextClassificationDataset } from "@/models/TextClassification";
 import { TokenClassificationDataset } from "@/models/TokenClassification";
+import { Vector } from "@/models//Vector";
```
Comment (frascuchon) [inline frontend/database/index.js]: @/models/Vector ?

File: frontend/database/modules/datasets.js [hunk 2]
```diff
@@ -631,9 +636,58 @@ const actions = {
...
 };
 
+const initVectorModel = (dataset) => {
+  deleteAllVectorData();
+  const records = dataset.results?.records;
+  const isDatasetContainsAnyVectors = isAnyKeyInArrayItem(records, "vectors");
+  if (isDatasetContainsAnyVectors) {
+    records.forEach(({ id: recordId, vectors }) => {
+      insertVectorModel(dataset.id, recordId, vectors);
+    });
```
Comment (frascuchon) [inline frontend/database/modules/datasets.js]: Here, could we insert data in batch just passing the info for all records at once?

File: frontend/models/Vector.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class Vector extends Model {
+  static entity = "vectors";
+  static primaryKey = ["vector_name", "dataset_id", "record_id"];
+
+  static fields() {
+    return {
+      id: this.uid(),
+      vector_name: this.attr(null),
+      vector_values: this.attr([]),
+      is_reference: this.boolean(false),
```
Comment (frascuchon) [inline frontend/models/Vector.js]: what's the purpose of this field?

**Review by frascuchon — APPROVED** [review]
It looks fine. Just some minimal comments.

File: frontend/models/Vector.js (hunk 3 above)
Comment (keithCuniah) [reply frontend/models/Vector.js]: After the last discussion the parameter is removed

**Review by leiyre — APPROVED** [review]

**Pull request merged**